lib/krb5/cache.c

```diff
@@ -252,6 +252,42 @@ krb5_cc_resolve(krb5_context context,
     return allocate_ccache(context, &krb5_fcc_ops, name, NULL, id);
 }
 
+#ifdef _WIN32
+static const char *
+get_default_cc_type_win32(krb5_context context)
+{
+    krb5_error_code ret;
+    krb5_ccache id;
+
+    /*
+     * If the MSLSA ccache type has a principal name,
+     * use it as the default.
+     */
+    ret = krb5_cc_resolve(context, "MSLSA:", &id);
+    if (ret == 0) {
+        krb5_principal princ;
+        ret = krb5_cc_get_principal(context, id, &princ);
+        krb5_cc_close(context, id);
+        if (ret == 0) {
+            krb5_free_principal(context, princ);
+	    return "MSLSA";
+	}
+    }
+
+    /*
+     * If the API:krb5cc ccache can be resolved,
+     * use it as the default.
+     */
+    ret = krb5_cc_resolve(context, "API:krb5cc", &id);
+    krb5_cc_close(context, id);
+    if (ret == 0) {
+	return "API"; /* XXX will this work? */
```

Comment:
Yes, it should work.  the `"API:krb5cc"` might not be needed -- `"API:"` alone should work fine, I think, since `acc_get_default_name()` does do something to discover a proper default ccache name (namely it calls the CCAPI's `get_default_ccache_name()` function).